### ISSUE TITLE
Temporary direct file sharing links

### ISSUE THREAD

**Issue by adx442** [jellyfin/jellyfin#808]
It would be really useful to be able to create direct, no authentication, time limited direct access links to a single content item.  The idea would be to e-mail or text a direct link to a piece of content that allows the recipient to click the link and play it without having an account on the server or allowing a "no restrictions" library.  The link would be configurable to expire after x plays or x days.  

Use case:  I'd like to send a home video of my kid's X-mas present opening to several friends, without giving them an account on my server, and be able to take advantage of the transcoding capability so that they can simply play it in their browser by clicking the link.

Emby had no interest in implementing this one, and I believe that decision was mostly based on their planned business model ... maybe the Jellyfin devs feel differently!

**Comment by kai-cool-dev** [comment]
+ 1

**Comment by onny** [comment]
I would also like to see this feature for playlist sharing :+1:

**Comment by MichaelFay** [comment]
I would also enjoy this feature

**Comment by mattthhdp** [comment]
i think the easiest way would be to use in a container.
https://gitlab.com/mojo42/Jirafeau 
Jirafeau do exactly what you want.

**Comment by JustAMan** [comment]
I thought Jirafeau is about sharing files for download, not about viewing them online. That said, this is a service I myself want to start hosting, too, though not for this exact purpose :)

**Comment by adx442** [comment]
> i think the easiest way would be to use in a container.
> https://gitlab.com/mojo42/Jirafeau
> Jirafeau do exactly what you want.

Not what I had in mind, I could always send someone a Nextcloud link if I just wanted to transfer the entire file.  What I was hoping for is to take advantage of the realtime transcoding to deliver an appropriately sized, network speed adjusted copy of the video to watch without making them a permanent account.  E.g., they click the direct link (hopefully one time use) and it plays, without a sign in required.  

A good example would be our home videos from this Christmas.  Shot and stored in 4K for editing and archiving, but a direct Jellyfin link would serve it out to family and friends at 720P without making accounts for 5 people that I don't want using my server in general.

**Comment by shmilyoo** [comment]
+1
When I just want to temporarily share a media, I don't want to create a new user specifically for this demand, or give my user account directly to others.

The link can:
* With permission control, restrict the user's use time, and even allow only one access process;
* Limit the number of times users can share this link with others:

**Comment by JustAMan** [comment]
Please vote and add comments here: https://features.jellyfin.org/posts/72/temporary-direct-file-sharing-links

**Comment by loligans** [comment]
@joshuaboniface not sure who else to reach out to, but github said you're the project lead. i'd be happy to take a stab at this, but I want to work with someone on the Jellyfin team to make sure i don't do a bunch of work to only get rejected. also would need to flesh out the project requirements and goals.

**Comment by crobibero** [comment]
@loligans best way to contact the team is through matrix. 

https://jellyfin.org/contact/

Join the #jellyfin-dev room

**Comment by loligans** [comment]
to update those who aren't in the matrix chat. we talked about this feature a little bit. it's not ready for any sort of work yet because some database work needs to be done with API keys. i'm over simplifying here but basically the database would need to be changed to give API Keys permissions on a per entity basis. 

TLDR: the database needs changes to support creating an API key that has a `view` permission only for that christmas video you wanna show greg at the office and nothing else.

i'm going to keep my eye on this but i don't think this feature work will be ready to start until jellyfin 11 at the very least, but when it is i'll jump on it.

**Comment by timonoj** [comment]
Hi...Just checking. Has there been any further developments on this in the last 10 months? As I'd really like to be able to temporarily share a single movie with all the transcoding easiness Jellyfin provides.

**Comment by MagicLoustic** [comment]
+1

**Comment by userliluzibert** [comment]
I would also love to have this feature. I've been using Synology Video Station for a while now, and they have public link sharing with the ability to toggle the link on and off after the user is finished. It would be amazing for Jellyfin to offer this feature, or something similar to it as well.

**Comment by realies** [comment]
Add this please.

**Comment by r0bbie** [comment]
Regularly checking in on this one to see if any progress - very hopefully to see it be marked as officially "planned" soon...

To add to what others have said, my ideal implementation on this would be as seamless as possible:

\- "Share" button added to individual media pages  
\- Generate share links with options to   
          - allow non logged in users to view  
          - add expiry date  
          - require link password for non logged in users   
\- "Share link" capability enabled by admin on per-user basis?

Nextcloud's share functionality offers a pretty good example of how this could work imo.
